waddrmgr/db.go

```diff
@@ -862,14 +1225,108 @@ func existsAddress(tx walletdb.Tx, addressID []byte) bool {
 	return bucket.Get(addrHash[:]) != nil
 }
 
+// fetchAddrAccount returns the account to which the given address belongs to.
+// It looks up the account using the addracctidx index which maps the address
+// hash to it's corresponding account id.
+func fetchAddrAccount(tx walletdb.Tx, addressID []byte) (uint32, error) {
+	bucket := tx.RootBucket().Bucket(addrAcctIdxBucketName)
+
+	addrHash := fastsha256.Sum256(addressID)
+	val := bucket.Get(addrHash[:])
+	if val == nil {
+		str := "account not found"
+		return 0, managerError(ErrAccountNotFound, str, nil)
+	}
+	return binary.LittleEndian.Uint32(val), nil
+}
+
+// fetchAccountAddresses loads information about addresses of an account from the database.
+// The returned value is a slice address rows for each specific address type.
+// The caller should use type assertions to ascertain the types.
+func fetchAccountAddresses(tx walletdb.Tx, account uint32) ([]interface{}, error) {
```

Comment:
Fair point.  It's only being used for the error message though.  How about just removing that parameter from the function calls and error messages, and then, at the two call sites for `fetchAddress`:

```Go
if err != nil {
	if merr, ok := err.(*ManagerError); ok {
		desc := fmt.Sprintf("failed to fetch address '%s': %v",
			address.ScriptAddress(), merr.Description)
		merr.Description = desc
	}
}
```

Then add a comment on the errors in the functions that they are later prefixed with the address that caused the failure.